picard/ui/metadatabox/tagdiff.py

```diff
@@ -342,3 +349,41 @@ def update_tag_names(self, changes_first=False, top_tags=None):
             self.tag_names = [
                 tag for tag in tag_names if
                 self.status[tag] != TagStatus.EMPTY]
+
+    def to_json(self):
+        result = {}
+        for tag in self.tag_names:
+            result[tag] = {}
+            if tag in self.old:
+                result[tag][self.OLD_VALUE] = self.old[tag]
+            if tag in self.new:
+                result[tag][self.NEW_VALUE] = self.new[tag]
+
+        return json.dumps(result)
+
+    def handle_length(self, value, prettify_times=False):
+        try:
+            value = value if isinstance(value, str) else ''
+            return [format_time(value or 0)] if prettify_times else [value]
+        except (TypeError, ValueError) as e:
+            log.warning(e)
+            return ['']
+
+    def to_tsv(self, prettify_times=True):
+        f = io.StringIO()
+        writer = csv.writer(f, dialect='excel-tab')
+
+        for tag in self.tag_names:
+            # Take care with the special case of '~length', as that is not wrapped in a list it's a plain string
+            old = self.old[tag]
+            new = self.new[tag]
```

Comment:
Move those statements to an `else` branch below

Reply:
I'm not sure about that, I find this more readable. This highlights the special case (rather than 'alternative' case) for lengths. To put it another way, if we have a future PR that returns `~length` as an array with a single element (to match other tags) then the three lines  for `~length` would just be removed